### PR TITLE
Fix Issue [Bug] Login Page - Anki Icon Flickers (#8354)

### LINKED ISSUE #8354: [Bug] Login Page - Anki Icon Flickers
###### Reproduction Steps

1. Open Login Page (vertical screen)
2. Anki Icon appears

###### Expected Result

* Anki Icon stays on screen (it's vertical, and my phone has room for the icon).

###### Actual Result

* Anki Icon then quickly disappears

###### Debug info
```
AnkiDroid Version = 2.15alpha32

Android Version = 11

ACRA UUID = 2775070f-9c41-4dc3-886f-63fc5769ad8f

Scheduler = std2

Crash Reports Enabled = false

DatabaseV2 Enabled = true
```

###### Research
- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by bhamrickatmills**
## Purpose / Description

Fix  the occasional screen flicker which occurred when the login page would load for the first time.

## Fixes

Fixes #8354

## Approach

Removed onOrientationChangedListener and added onConfigurationChanged().

## How Has This Been Tested?

Tested on physical devices:
Kindle fire 7 tablet
Fire OS 7.3.1.8

Samsung-SM-J327V phone
Android version 8.1.0

Tested on emulated devices:
Pixel 3 API 30 phone
Pixel C API 30 tablet

## Learning (optional, can help others)
https://stackoverflow.com/questions/5015094/how-to-determine-device-screen-size-category-small-normal-large-xlarge-usin

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements)
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by david-allison-1 — DISMISSED** [review]
When I rotate my phone, the logo doesn't disappear.

Then when I wobble my phone, it flickers in and out of view.

----

This seems to conflate the orientation of the screen with the rotation of the screen.

It seems to me that this should mirror the phone's "rotation" status (unless there's a reason not to).

We can handle a change in the phone's rotation status via `onConfigurationChanged` (I've just tested this and it seems to work)

https://stackoverflow.com/questions/9566633/how-to-use-onconfigurationchanged-and-newconfig-orientation-configuration-o

https://developer.android.com/reference/android/app/Activity#onConfigurationChanged(android.content.res.Configuration)

After we detect a change, we need to know: 

* Was there a change in orientation
* If so: "do we have enough room to display the logo" 
    * I expect that this if we use this strategy, it shouldn't matter if the display is square, or if we can't determine if the status has changed.

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java [hunk 1]
```diff
@@ -340,75 +315,22 @@ protected String getHumanReadableLoginErrorMessage(Exception exception) {
...
-    }
-
-
-    @Override
-    protected void onPause() {
-        super.onPause();
-        myOrientationEventListener.disable();
+    public void onConfigurationChanged(@NonNull Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+        if(isScreenSmall() && newConfig.orientation == Configuration.ORIENTATION_LANDSCAPE){
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java:333]: I'm not sure if `isScreenSmall()` will be valid here - `480x640` might not be sufficient.

It works on my screen, I'll do a little more exploratory testing.

```suggestion
        if (isScreenSmall() && newConfig.orientation == Configuration.ORIENTATION_LANDSCAPE) {
```

**Review by david-allison-1 — COMMENTED** [review]
It's lovely to see that much code being removed, especially for better functionality, kudos to you!

I'd like to have a little more time playing with emulators to see if `isScreenSmall()` works for all devices via emulation.

It'd be useful if you could take a look as well

But, the code's great and works well! Thanks

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java (hunk 1 above)
Comment (bhamrickatmills) [reply AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java:333]: Spacing... oops! I'll fix that and play with the emulators. And thank you for the feedback, and the recommendation to use onConfigurationChanged()!

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java:333]: No worries! We have an extensive code style: https://github.com/ankidroid/Anki-Android/wiki/Code-style I'm working on automating things so we won't need to worry about it in reviews

**Review by david-allison-1 — APPROVED** [review]
Looks great!

**Review by mikehardy — APPROVED** [review]
certainly an easier way of doing it vs the listener before - a shame that did not work as expected, but just listening for configuration changes appears quite effective and very clean as implementation

thanks!

**Comment by mikehardy** [comment]
Hi there!

Just a friendly notice (one per person for PRs merged in March, regardless of PR quantity) that we try to process OpenCollective payments monthly - it's time for March 2021 submissions

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

Thanks!

**Pull request merged**